lib/active_model/serializer/adapter/json.rb

```diff
@@ -13,27 +18,16 @@ def serializable_hash(options = nil)
               serializer.attributes(options)
             end
 
-            serializer.associations.each do |association|
-              serializer = association.serializer
-              opts = association.options
-
-              if serializer.respond_to?(:each)
-                array_serializer = serializer
-                hash[association.key] = array_serializer.map do |item|
-                  cache_check(item) do
-                    item.attributes(opts)
-                  end
+            included_associations = serializer.expand_includes(@included)
+            included_associations.each do |association, assoc_includes|
+              hash[association.key] =
+                if association.options[:virtual_value]
+                  association.options[:virtual_value]
+                elsif association.serializer && association.serializer.object
+                  FlattenJson.new(association.serializer,
+                                  association.options.merge(include: assoc_includes))
```

Comment:
This line gives a rubocop warning about aligning method parameters that span multiple lines, although I believe the indentation is right here.